index.js

```diff
@@ -47,12 +47,28 @@ function requestbody(opts) {
   opts.formidable = 'formidable' in opts ? opts.formidable : {};
   opts.includeUnparsed = 'includeUnparsed' in opts ? opts.includeUnparsed : false
   opts.textLimit = 'textLimit' in opts ? opts.textLimit : '56kb';
-  opts.strict = 'strict' in opts ? opts.strict : true;
+
+  // @todo: next major version, opts.strict support should be removed
+  if (opts.strict && opts.parsedMethods) {
+    throw new Error('Cannot use strict and parsedMethods options at the same time.')
+  }
+
+  if ('strict' in opts) {
+    console.warn('DEPRECATED: opts.strict has been deprecated in favor of opts.parsedMethods.')
+    if (opts.strict) {
+      opts.parsedMethods = ['POST', 'PUT', 'PATCH']
+    } else {
+      opts.parsedMethods = ['POST', 'PUT', 'PATCH', 'GET', 'HEAD', 'DELETE']
```

Comment:
My only concern with this entire PR is that we might be making a false assumption that users that set `opts.strict` to `false` only expect koa-body to parse on POST, PUT, PATCH, GET, HEAD, and DELETE methods.

This might be a problem if a user sets `opts.strict` = `false` and a custom method, e.g. `OPTIONS` or `TRACE` and expects a parsed body and then updates to this version.

Please let me know if I'm thinking about this correctly.

Reply:
One thought I had was to roll this and a few other changes into a major version and worry less about these edge-cases.

Reply:
While yes, that's definitely an edge case, we're entering edge case saloon territory; `OPTIONS` currently has no realizable use for a request body, and `TRACE` actually specifies in RFC 7231 that it "MUST NOT" possess a body.

Reply:
Ok, makes sense. I think that this is an acceptable risk with a minor version.